zfs/zfs.go

```diff
@@ -683,6 +652,76 @@ func (a ZFSSendArgsUnvalidated) Validate(ctx context.Context) (v ZFSSendArgsVali
 	}, nil
 }
 
+func (a ZFSSendArgsUnvalidated) buildCommonSendArgs() []string {
+
+	args := make([]string, 0)
+
+	if a.Raw {
+		args = append(args, "-w")
+	}
+
+	if a.Properties {
+		args = append(args, "-p")
+	}
+
+	if a.BackupProperties {
+		args = append(args, "-b")
+	}
+
+	if a.LargeBlocks {
+		args = append(args, "-L")
+	}
+
+	if a.Compressed {
+		args = append(args, "-c")
+	}
+
+	if a.EmbeddedData {
+		args = append(args, "-e")
+	}
+
+	return args
+}
+
+// ResumeToken is allowed to be empty
+// a must already be validated
+//
+// SECURITY SENSITIVE because Encrypted must be handled correctly
+func (a ZFSSendArgsValidated) buildSendArgs() ([]string, error) {
+	args := a.buildCommonSendArgs()
```

Comment:
restore empty list initialisation and move buildCommonSendArgs below the token resume stuff?

Reply:
Yes, if I understand correctly that's what I mean by https://github.com/zrepl/zrepl/pull/285#discussion_r484458572